configure.sh

```diff
@@ -13,167 +13,235 @@
+  cat > third_party/python_legacy/defs.bzl <<EOF
+# AUTOGENERATED by configure.sh
+load("@bazel_tools//tools/python:toolchain.bzl", "py_runtime_pair")
+# Absolute path to the python interpreter Bazel/TF should use:
+interpreter = "${PYTHON_BIN_PATH}"
+py_runtime = native.py_runtime
+EOF
+
+  echo "Wrote third_party/python_legacy with interpreter=${PYTHON_BIN_PATH}"
 }
 
+# --- start fresh -----------------------------------------------------------
+rm -f .bazelrc .tf_configure.bazelrc
 
-# Remove .bazelrc if it already exist
-[ -e .bazelrc ] && rm .bazelrc
-
-# Check if we are building GPU or CPU ops, default CPU
-while [[ "$TF_NEED_CUDA" == "" ]]; do
-  read -p "Do you want to build ops again TensorFlow CPU pip package?"\
-" Y or enter for CPU (tensorflow-cpu), N for GPU (tensorflow). [Y/n] " INPUT
-  case $INPUT in
-    [Yy]* ) echo "Build with CPU pip package."; TF_NEED_CUDA=0;;
-    [Nn]* ) echo "Build with GPU pip package."; TF_NEED_CUDA=1;;
-    "" ) echo "Build with CPU pip package."; TF_NEED_CUDA=0;;
-    * ) echo "Invalid selection: " $INPUT;;
+# --- parse args ------------------------------------------------------------
+USER_PY=""
+for arg in "$@"; do
+  case "$arg" in
+    --python=*) USER_PY="${arg#--python=}" ;;
+    *) echo "Unknown arg: $arg" ;;
   esac
 done
 
-while [[ "$TF_CUDA_VERSION" == "" ]]; do
-  read -p "Are you building against TensorFlow 2.1(including RCs) or newer?[Y/n] " INPUT
-  case $INPUT in
-    [Yy]* ) echo "Build against TensorFlow 2.1 or newer."; TF_CUDA_VERSION=11;;
-    [Nn]* ) echo "Build against TensorFlow <2.1."; TF_CUDA_VERSION=10.0;;
-    "" ) echo "Build against TensorFlow 2.1 or newer."; TF_CUDA_VERSION=11;;
-    * ) echo "Invalid selection: " $INPUT;;
+# --- choose interpreter (venv/conda/system) --------------------------------
+if [[ -n "${USER_PY}" ]]; then
+  # 1) Explicit --python=... flag
+  PY="${USER_PY}"
+elif [[ -n "${PYTHON_BIN_PATH:-}" ]]; then
+  # 2) Explicit environment override
+  PY="${PYTHON_BIN_PATH}"
+elif [[ -n "${CONDA_PREFIX:-}" && -x "${CONDA_PREFIX}/bin/python" ]]; then
+  # 3) Conda environment python, if available
+  PY="${CONDA_PREFIX}/bin/python"
+else
+  # 4) Fallback: system python3, but require >= 3.10
+  if ! command -v python3 >/dev/null 2>&1; then
+    die "python3 not found. Pass --python=/path/to/python3.10+ or set PYTHON_BIN_PATH."
+  fi
+
+  if ! python3 - <<'PY'
+import sys
+raise SystemExit(0 if sys.version_info[:2] >= (3, 10) else 1)
+PY
+  then
+    die "Python 3.10+ required for TensorFlow Quantum; found $(python3 -V 2>&1). Pass --python=/path/to/python3.10+ or set PYTHON_BIN_PATH."
+  fi
+
+  PY="$(command -v python3)"
+fi
+
+# Normalize to an absolute path (readlink -f is GNU; fall back to python)
+if command -v readlink >/dev/null 2>&1; then
+  PY_ABS="$(readlink -f "${PY}" 2>/dev/null || true)"
+fi
+if [[ -z "${PY_ABS:-}" ]]; then
+  PY_ABS="$("${PY}" - <<'PY'
+import os,sys
+print(os.path.abspath(sys.executable))
+PY
+)"
```

Comment:
Although this works, IMHO this construct is fragile because of the complicated syntax. (At first, I thought the `)"` by itself at the end was a typo.) Would the following not work?

```suggestion
  PY_ABS="$("${PY}" -c 'import os,sys; print(os.path.abspath(sys.executable))')"
```